File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
import sys
input=sys.stdin.readline
def check(s,l,r):
    if l>=r:
        return 1,1
    if s[l]!=s[r]:
        return 0,1
    res,c=check(s,l+1,r-1)
    return res,c+1

t=int(input())
for _ in range(t):
    s=input().strip()
    r,c=check(s,0,len(s)-1)
    print(r)
    print(c)

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.java
import java.io.*;
public class Main {
    static int cnt;
    static int check(String s, int l, int r){
        cnt++;
        if(l>=r) return 1;
        if(s.charAt(l)!=s.charAt(r)) return 0;
        return check(s,l+1,r-1);
    }
    public static void main(String[] args) throws Exception{
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        int t=Integer.parseInt(br.readLine().trim());
        StringBuilder sb=new StringBuilder();
        while(t-->0){
            String s=br.readLine().trim();
            cnt=0;
            int res=check(s,0,s.length()-1);
            sb.append(res).append(" ").append(cnt).append("\n");
        }
        System.out.print(sb.toString());
    }
}